Add Trino 450 release notes (#22327)
trinodb/trino: docs/src/main/sphinx/release/release-450.md

> # Release 450 (12 Jun 2024)

Reviewer: 18

## General

* Add support for defining a number of rows to be pushed from a filter into
  values with the `optimizer.push-filter-into-values-max-row-count`
  configuration property. ({issue}`22105`)
* Add support for specifying an Azure blob endpoint for accessing spooling in
  fault-tolerant execution with the `exchange.azure.endpoint` configuration
  property. ({issue}`22218`)
* Expose driver execution statistics via JMX. ({issue}`22427`)
* Improve performance of the {func}`first_value` and {func}`last_value`
  functions. ({issue}`22092`)
* Improve performance for large clusters under heavy workloads. ({issue}`22039`)
* Improve performance of queries with simple predicates. This optimization can
  be disabled using the `experimental.columnar-filter-evaluation.enabled`
  configuration property or the `columnar_filter_evaluation_enabled` session
  property. ({issue}`21375`)

> * {{breaking}} Improve performance of distinct aggregations, and replace the

Reviewer 2: "... of aggregations containing a `DISTINCT` clause ..."

  `optimizer.optimize-mixed-distinct-aggregations` configuration properties with
  the new `optimizer.distinct-aggregations-strategy` property. ({issue}`21907`)
* Improve performance of reading JSON files. ({issue}`22348`)
* Improve performance for the {func}`date_trunc`, {func}`date_add`, and
  {func}`date_diff` functions. ({issue}`22192`)
* Fix failure when loading the [](/admin/event-listeners-openlineage). ({issue}`22228`)

> * Fix potential incorrect results when a cached value has been removed or

Reviewer: ```suggestion
* Fix potential incorrect results when a cached value was removed or
```

> * Fix potential incorrect results when a cached value was removed or updated. ({issue}`22285`)

Reviewer 2: Cached where? What values?

Author: In many different places, and it depends... metadata caches for some connectors, regular data caches for others.

## Security

> * Add support for exclusively using web identity for authentication when using

Reviewer 2: This item is hard to parse and understand. How about:

"Add support for using web identity exclusively for authentication when running on Amazon EKS with the legacy S3 file system enabled. This can be configured via the `trino.s3.use-web-identity-token-credentials-provider` property."

  Amazon EKS with Kubernetes by setting the
  `trino.s3.use-web-identity-token-credentials-provider` configuration property. ({issue}`22162`)
* Add support for exclusively using web identity for authentication when using
  Amazon EKS with
  [IAM roles](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html)
  by setting the
  `s3.use-web-identity-token-credentials-provider` configuration property. ({issue}`22163`)

## JDBC driver

* Add support for the `assumeNullCatalogMeansCurrent` connection property. When
  enabled, a `null` value for the `catalog` parameter in `DatabaseMetaData`
  methods is assumed to mean the current catalog. If no current catalog is
  set, the behaviour is unmodified. ({issue}`20866`)

## BigQuery connector

> * Add support for name caching when the

Reviewer: name of what?

  `bigquery.case-insensitive-name-matching` configuration property is enabled. ({issue}`10740`)
* {{breaking}} Automatically configure BigQuery scan parallelism, and remove the
  `bigquery.parallelism` configuration property. ({issue}`22279`)

## Cassandra connector

> * Fix incorrect behavior that uses one more partition than expected when

Reviewer 2: What does "uses one more partition than expected mean"? Process a partition? Read a partition? Return results from a partition?

* Add support for using table statistics with `TIMESTAMP` types. ({issue}`21878`)
* Add support for reading tables with
  [type widening](https://docs.delta.io/latest/delta-type-widening.html). ({issue}`21756`)
* Set the default value for the `s3.max-connections` configuration property
  to 500. ({issue}`22209`)
* Fix failure when reading a `TIMESTAMP` value after the year 9999. ({issue}`22184`)
* Fix failure when reading tables with the unsupported `variant` type. ({issue}`22310`)

## Hive connector

* Add support for `integer` to `varchar` and `decimal` to `varchar` type 
  coercion in unpartitioned tables. ({issue}`22246`, {issue}`22293`)
* Add support for `double` to `varchar` type coercion in unpartitioned tables
  using Parquet files. ({issue}`22277`)
* Add support for `float` to `varchar` type coercion. ({issue}`22291`)
* Set the default value for the `s3.max-connections` configuration property
  to 500. ({issue}`22209`)

## Hudi connector

* Set the default value for the `s3.max-connections` configuration property
  to 500. ({issue}`22209`)

## Iceberg connector

* Add support for reading Iceberg

>   [UniForm](https://docs.delta.io/latest/delta-uniform.html) tables. ({issue}`22106`)

Reviewer: Is it really "Iceberg Uniform" or just "Uniform"

* Add support for the `TRUNCATE` statement. ({issue}`22340`)

> * Add support for V2 of the Nessie REST API. Previous behavior can be

Reviewer: So its actually .. Change to use the V2 protocol by default..

Author: We didn't previously support V2, did we?

Reviewer: no .. but we are not only adding support .. we are switching as well..

Reviewer: so maybe that is also a breaking change?

  restored by setting the `iceberg.nessie-catalog.client-api-version`
  configuration property to `V1`. ({issue}`22215`)
* Improve performance when reading by populating `split_offsets` in file
  metadata. ({issue}`9018`)
* Set the default value for the `s3.max-connections` configuration property
  to 500. ({issue}`22209`)
* Fix failure when reading Parquet files that don't have `field-id` on
  structured types. ({issue}`22347`)

## MariaDB connector

* Add support for [fault-tolerant execution](/admin/fault-tolerant-execution). ({issue}`22328`)
* Improve performance of listing table columns. ({issue}`22241`)

## Memory connector

> * Add support for the `TRUNCATE` statement. ({issue}`22337`)

Reviewer: Nice!

## MySQL connector

* Improve performance of listing table columns. ({issue}`22241`)

## Pinot connector

* Add support for the
  [`enableNullHandling` query option](https://docs.pinot.apache.org/developers/advanced/null-value-support#advanced-null-handling-support). ({issue}`22214`)

> * Fix failure when using dynamic tables. ({issue}`22301`)

Reviewer: Those are Pinot dynamic tables I assume ... should we link to some docs or so (I couldnt find any quickly)

# Release 450 (7 Jun 2024)

## General

> * Add support for the `optimizer.push-filter-into-values-max-row-count`

Reviewer: Invert 

Add support to define the number of rows to push from a filter into a value with the xx prop

> * Add support for defining a number of rows to be pushed from a filter into

Reviewer 2: This should describe the user-visible impact. "pushed from a filter into values" doesn't mean anything for an end user.

Author: I'm honestly not too sure what the feature is doing, and it wasn't documented, so there's not much to fall back on 😔

Reviewer: Should we reword around performance and make it higher level?

Reviewer 2: I think we should remove this item. It's too low level and doesn't have much direct visible impact to end users.

  configuration property, which can define a number of rows to be pushed from a
  filter into values. ({issue}`22105`)
* Improve performance of the {func}`first_value` and {func}`last_value`
  functions. ({issue}`22092`)

> * Improve performance for large clusters under heavy workloads. ({issue}`22039`)

Reviewer: super vague ..

Author: I'm not really sure how else to word it...

Reviewer: yeah .. fine to leave as is

> * Improve performance of queries with simple predicates. This optimization can

Reviewer: is "simple predicates" specific enough? I kinda think yes and people can look up more if needed. Also .. is this documented? And "experimental" .. do we need to mention that besides the name of the property?

  be disabled using the `experimental.columnar-filter-evaluation.enabled`

>   configuration property or `columnar_filter_evaluation_enabled` session

Reviewer: or the

  property. ({issue}`21375`)

> * Fix failure when loading the {doc}`/admin/event-listeners-openlineage`. ({issue}`22228`)

Reviewer: use markdown syntax 

```suggestion
* Fix failure when loading the [](/admin/event-listeners-openlineage). ({issue}`22228`)
```

## JDBC driver

> * Add support for the `assumeNullCatalogMeansCurrent` connection property. When

Reviewer: fyi .. this is not documented on purpose since it seems to be super niche problem

  enabled, a `null` value for the `catalog` parameter in `DatabaseMetaData`
  methods is assumed to mean the current catalog. If no current catalog is
  set, the behaviour is unmodified. ({issue}`20866`)

## BigQuery connector

> * Add support for caching when the `bigquery.case-insensitive-name-matching`

Reviewer: caching what?

Reviewer: and how is that related to the property?

  configuration property is enabled. ({issue}`10740`)

## Cassandra connector

* Fix incorrect behavior that uses one more partition than expected when
  specifying a value for the `cassandra.partition-size-for-batch-select`
  configuration property. ({issue}`21940`)

## Delta Lake connector

* Add support for concurrent `UPDATE`, `MERGE`, and `DELETE` queries. ({issue}`21727`)
* Fix failure when reading a `TIMESTAMP` value after the year 9999. ({issue}`22184`)

## Hive connector

* Add support for `integer` to `varchar` and `decimal` to `varchar` type 
  coercion in unpartitioned tables. ({issue}`22246`, {issue}`22293`)
* Add support for `double` to `varchar` type coercion in unpartitioned tables
  using Parquet files. ({issue}`22277`)
* Add support for `float` to `varchar` type coercion. ({issue}`22291`)

## MariaDB connector

> * Improve performance of listing table columns. ({issue}`22241`)

Reviewer: This does not affects SQL Server, PostgreSQL or Oracle?

Author: https://github.com/trinodb/trino/pull/22241 seems to only list those four

Reviewer: ok
